core/src/main/scala/org/broadinstitute/dsp/workbench/welder/model.scala

```diff
@@ -28,10 +31,27 @@ object SyncStatus {
 
   val stringToSyncStatus: Set[SyncStatus] = sealerate.values[SyncStatus]
 }
-final case class BucketNameAndObjectName(bucketName: GcsBucketName, blobName: GcsBlobName) {
-  override def toString: String = s"${bucketName.value}/${blobName.value}"
+
+sealed abstract class SyncMode extends Product with Serializable
+object SyncMode {
+  final case object SafeMode extends SyncMode {
+    override def toString: String = "SAFE"
+  }
+  final case object Edit extends SyncMode {
+    override def toString: String = "EDIT"
+  }
 }
 
+sealed abstract class SourceUri
+object SourceUri {
+  final case class DataUri(data: Array[Byte]) extends SourceUri
+  final case class GsPath(bucketName: GcsBucketName, blobName: GcsBlobName) extends SourceUri {
+    override def toString: String = s"gs://${bucketName.value}/${blobName.value}"
+  }
+}
+
+final case class StorageLink(localBaseDirectory: Path, cloudStorageDirectory: GsPath, pattern: String, recursive: Boolean)
```

Comment:
should we also move `StorageLinks` and the encoders/decoders here?

Reply:
sounds good...